### ISSUE TITLE
Support for multiple persistent mounts 

### ISSUE THREAD

**Issue by astigmatism** [caiiiycuk/js-dos.com#70]
I've discovered you can only invoke the DosFS.extract() method once with a persistentMount parameter. Attempting to do so again enters the code path with the message "Can't create persistent mount point, after syncing process starts" where the promise is rejected. However, I believe it could be beneficial to have multiple persistent mounts:

In my case, I'd like to extract the Gravis Ultrasound patch files to "/ultrasnd" (as the dosbox conf suggests) in addition to a game at another persistentMount location (eg /digger)

**Comment by caiiiycuk** [comment]
I see, good point!

**Comment by caiiiycuk** [comment]
Btw, could be solution for you to use subfolder, like:
* /xxx/ultrasnd
* /xxx/game

So you will mount only /xxx

**Comment by astigmatism** [comment]
I realize this but it also means duplicating the /xxx/ultrasnd files for
all games which utilize them. This is not very efficient.

Eric

**Comment by caiiiycuk** [comment]
I think to change api in that way this:
```
fs.extract([
{ url: "file1.zip", path: "/path1"},
{ url: "file2.zip", path: "/path2"},
//...
])
```

@astigmatism, @ZaDarkSide are you ok with this API?

**Comment by ZaDarkSide** [comment]
Yes, it looks good to me! Thank you!

**Comment by caiiiycuk** [comment]
also I can add caching for files:
```
fs.extract({url:"...", path: "...", version: "xxx"})
```
so it will be cached by js-dos and changed when file version is changed

and no version - no cache

**Comment by ZaDarkSide** [comment]
That sounds perfect, it will save bandwidth to not download the same zip if it is cached already.

**Comment by caiiiycuk** [comment]
Fixed in 6.22.51, now you can use multiple persistent point at once.
http://js-dos.com/#js-dos-622-faq-using-multiple-archives-at-once

Btw, I didn't add explicit cache, because of nature of persistent folder. Archives downloaded only once and then indexed db version is used (to avoid resetting of persistent content)

**Comment by astigmatism** [comment]
Very much appreciated! Great work :)
